lib/suites/LinkedDataSignature.js

```diff
@@ -11,37 +11,57 @@ const LinkedDataProof = require('./LinkedDataProof');
 
 module.exports = class LinkedDataSignature extends LinkedDataProof {
   /**
-   * @param type {string} Provided by subclass.
+   * @param {object} options - Options hashmap.
+   * @param {string} options.type - Suite name, provided by subclass.
+   * @typedef LDKeyPair
+   * @param {LDKeyPair} LDKeyClass - The crypto-ld key class that this suite
+   *   will use to sign/verify signatures. Provided by subclass. Used
+   *   during the `verifySignature` operation, to create an instance (containing
+   *   a `verifier()` property) of a public key fetched via a `documentLoader`.
    *
-   * @param [verificationMethod] {string} A key id URL to the paired public key.
+   * Either a `key` OR at least one of `signer`/`verifier` is required:
+   *
+   * @param {object} [options.key] - An optional key object (containing an
+   *   `id` property, and either `signer` or `verifier`, depending on the
+   *   intended operation. Useful for when the application is managing keys
+   *   itself (when using a KMS, you never have access to the private key,
+   *   and so should use the `signer` param instead).
+   * @param {Function} [options.signer] - Signer function that returns an
+   *   object with an async sign() method. This is useful when interfacing
+   *   with a KMS (since you don't get access to the private key and its
+   *   `signer()`, the KMS client gives you only the signer function to use).
+   * @param {Function} [options.verifier] - Verifier function that returns
+   *   an object with an async `verify()` method. Useful when working with a
+   *   KMS-provided verifier function.
    *
    * Advanced optional parameters and overrides:
    *
-   * @param [proof] {object} a JSON-LD document with options to use for
-   *   the `proof` node (e.g. any other custom fields can be provided here
-   *   using a context different from security-v2).
-   * @typedef {Object} LDKeyPair
-   * @param {LDKeyPair} LDKeyClass - The crypto-ld key type that this suite
-   *   will use to sign/verify signatures.
-   * @param [date] {string|Date} signing date to use if not passed.
-   * @param {function} [signer] - Signer function.
-   * @param [useNativeCanonize] {boolean} true to use a native canonize
-   *   algorithm.
+   * @param {string} [options.verificationMethod]  A key id URL of the paired
```

Comment:
Can we say when this would be passed in? It would only be used on the verification side if you've retrieved the verification method previously and don't need it to be looked up using the `documentLoader`, right?

Reply:
@dlongley hmm so honestly, I can't think of when it would be useful, on the verification side. Because if you've already retrieved it, the easiest thing is to just pass it in as a `key` param.